jmh-core/src/main/java/org/openjdk/jmh/generators/core/Paddings.java

```diff
@@ -25,34 +25,37 @@
 package org.openjdk.jmh.generators.core;
 
 import java.io.PrintWriter;
-import java.util.ArrayList;
-import java.util.List;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.StringJoiner;
 
 public class Paddings {
 
-    private static final List<String> PADDING_CACHE;
+    private static final Map<String, String> PADDING_CACHE =
+            Collections.synchronizedMap(new HashMap<>());
 
-    static {
-        PADDING_CACHE = new ArrayList<>();
+    private static String generate(String prefix) {
+        if (prefix.isEmpty()) {
+            throw new IllegalArgumentException("prefix must not be empty");
+        }
+        StringJoiner sj = new StringJoiner("");
```

Comment:
What's the point of rewriting this to `StringJoiner`? `StringBuilder` optimizations would be nice to have, if available in the JIT.

Reply:
All right, fine. But you need to pass GHA testing with this patch.